### PR TITLE
Support for OIDC scopes

### PR DESCRIPTION
**Pull request by afabiani**
…configuration

(cherry picked from commit 59b7b63294a3402b8652135451c5d8ca68d2baf9)

Fix https://github.com/geosolutions-it/MapStore2/issues/10151

### CODE REVIEW

File: src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/oauth2/DiscoveryClient.java [hunk 1]
```diff
@@ -103,22 +100,29 @@ public void autofill(OAuth2Configuration conf) {
...
+            if (conf.getScopes() == null || conf.getScopes().isEmpty()) {
+                Optional.ofNullable(response.get(getScopesSupported()))
+                        .ifPresent(
+                                s -> {
+                                    @SuppressWarnings("unchecked")
+                                    List<String> scopes = (List<String>) s;
+
+                                    // Get existing scopes from conf and split into a Set
+                                    Set<String> mergedScopes = new HashSet<>();
+                                    mergedScopes.addAll(Arrays.asList(conf.getScopes().split(",")));
```
Comment (offtherailz) [inline src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/oauth2/DiscoveryClient.java:114]: I have a `NullPointerException` when I do not set scopes.

```
	... 30 more
Caused by: java.lang.NullPointerException
	at it.geosolutions.geostore.services.rest.security.oauth2.DiscoveryClient.lambda$autofill$6(DiscoveryClient.java:114) ~[geostore-rest-impl-2.2-SNAPSHOT.jar:?]
	at java.util.Optional.ifPresent(Optional.java:183) ~[?:?]
	at it.geosolutions.geostore.services.rest.security.oauth2.DiscoveryClient.autofill(DiscoveryClient.java:107) ~[geostore-rest-impl-2.2-SNAPSHOT.jar:?]
	at it.geosolutions.geostore.services.rest.security.oauth2.openid_connect.OpenIdConnectFilter.<init>(OpenIdConnectFilter.java:71) ~[geostore-rest-impl-2.2-SNAPSHOT.jar:?]
	at it.geosolutions.geostore.services.rest.security.oauth2.openid_connect.OpenIdConnectSecurityConfiguration.oidcOpenIdFilter(OpenIdConnectSecurityConfiguration.java:145) ~[geostore-rest-impl-2.2-SNAPSHOT.jar:?]
	at it.geosolutions.geostore.services.rest.security.oauth2.openid_connect.OpenIdConnectSecurityConfiguration$$EnhancerBySpringCGLIB$$9ded83d5.CGLIB$oidcOpenIdFilter$3(<generated>) ~[geostore-rest-impl-2.2-SNAPSHOT.jar:?]
	at it.geosolutions.geostore.services.rest.security.oauth2.openid_connect.OpenIdConnectSecurityConfiguration$$EnhancerBySpringCGLIB$$9ded83d5$$FastClassBySpringCGLIB$$e4b2836b.invoke(<generated>) ~[geostore-rest-impl-2.2-SNAPSHOT.jar:?]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.3.18.jar:5.3.18]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331) ~[spring-context-5.3.18.jar:5.3.18]
	at it.geosolutions.geostore.services.rest.security.oauth2.openid_connect.OpenIdConnectSecurityConfiguration$$EnhancerBySpringCGLIB$$9ded83d5.oidcOpenIdFilter(<generated>) ~[geostore-rest-impl-2.2-SNAPSHOT.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
```

**Review by offtherailz — CHANGES_REQUESTED** [review]
Can not test. 
- [x] If I do not set scopes, it rises an exception (see inline comment)
- [ ] If I set scopes, `oidcOAuth2Config.scopes=openid,profile,groups`, I have a `No access token found on callback request. ` without any additional logging.

Scope should be optional, and when we have errors, we should have a proper logging to be able to identify the problem and possibly be able to fix the configuration issues, without the needing to inspect the code everytime. 
This issue without any logging happened a lot of times, and findi out the reason is extremely expensive in terms of time and resources,

**Comment by afabiani** [comment]
@offtherailz NPE fix pushed, for the second statement, rise the log level to DEBUG or check the Keycloack service log.

**Review by offtherailz — CHANGES_REQUESTED** [review]
- [ ] Logging for callback error (and errorDescription) in `/callback` endpoint
- [ ] Support for `groupMapping` as for keycloak

**Review by offtherailz — APPROVED** [review]

**Pull request merged**